spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -161,39 +162,68 @@ public TaskDefinitionResource display(@PathVariable("name") String name) {
 	 */
-	class Assembler extends ResourceAssemblerSupport<TaskDefinition, TaskDefinitionResource> {
+	class Assembler extends ResourceAssemblerSupport<TaskExecutionAwareTaskDefinition, TaskDefinitionResource> {
 
 		public Assembler() {
 			super(TaskDefinitionController.class, TaskDefinitionResource.class);
 		}
 
 		@Override
-		public TaskDefinitionResource toResource(TaskDefinition taskDefinition) {
-			return createResourceWithId(taskDefinition.getName(), taskDefinition);
+		public TaskDefinitionResource toResource(TaskExecutionAwareTaskDefinition taskExecutionAwareTaskDefinition) {
+			return createResourceWithId(taskExecutionAwareTaskDefinition.getTaskDefinition().getName(), taskExecutionAwareTaskDefinition);
 		}
 
 		@Override
-		public TaskDefinitionResource instantiateResource(TaskDefinition taskDefinition) {
-			String key = DeploymentKey.forTaskDefinition(taskDefinition);
-			String id = deploymentIdRepository.findOne(key);
-			boolean composed = taskService.isComposedDefinition(taskDefinition.getDslText());
-			TaskStatus status = null;
-			if (id != null) {
-				status = taskLauncher.status(id);
+		public TaskDefinitionResource instantiateResource(TaskExecutionAwareTaskDefinition taskExecutionAwareTaskDefinition) {
+			boolean composed = taskService.isComposedDefinition(taskExecutionAwareTaskDefinition.getTaskDefinition().getDslText());
+			TaskDefinitionResource taskDefinitionResource = new TaskDefinitionResource(taskExecutionAwareTaskDefinition.getTaskDefinition().getName(),
+					taskExecutionAwareTaskDefinition.getTaskDefinition().getDslText());
+			if(taskExecutionAwareTaskDefinition.getLatestTaskExecution() != null) {
+				taskDefinitionResource.setLastTaskExecution(new TaskExecutionResource(taskExecutionAwareTaskDefinition.getLatestTaskExecution()));
 			}
-			String state = (status != null) ? status.getState().name() : "unknown";
-			TaskDefinitionResource taskDefinitionResource = new TaskDefinitionResource(taskDefinition.getName(),
-					taskDefinition.getDslText());
 			taskDefinitionResource.setComposed(composed);
-			taskDefinitionResource.setStatus(state);
 			return taskDefinitionResource;
 		}
 	}
+
+	class TaskDefinitionConverter implements Converter<TaskDefinition, TaskExecutionAwareTaskDefinition> {
+		final Map<String, TaskExecution> taskExecutions;
+
+		public TaskDefinitionConverter(List<TaskExecution> taskExecutions) {
+			super();
+			if (taskExecutions != null) {
+				this.taskExecutions = new HashMap<>(taskExecutions.size());
+				for (TaskExecution taskExecution : taskExecutions) {
+					this.taskExecutions.put(taskExecution.getTaskName(), taskExecution);
+				}
+			}
+			else {
+				this.taskExecutions = null;
+			}
+		}
+
+		@Override
+		public TaskExecutionAwareTaskDefinition convert(TaskDefinition source) {
+
+			TaskExecution lastTaskExecution = null;
+
+			if (taskExecutions != null) {
+				lastTaskExecution = taskExecutions.get(source.getName());
+			}
+
+			if (lastTaskExecution != null) {
```

Comment:
A status of UKNOWN is not correct when working with a task that has not task-execution.  It should say "Not Executed"   Thoughts?

Reply:
Should probably be a follow up issue - Right now the statuses are mapped to what the UI supports.